DFC.ServiceTaxonomy.PageLocation/Drivers/PageLocationPartDisplayDriver.cs

```diff
@@ -49,15 +50,30 @@ public override async Task<IDisplayResult> UpdateAsync(PageLocationPart model, I
 
         private async Task ValidateAsync(PageLocationPart pageLocation, IUpdateModel updater)
         {
-            if (string.IsNullOrWhiteSpace(pageLocation.UrlName))
+            bool urlNameIsValid = true;
+
+            if (urlNameIsValid && string.IsNullOrWhiteSpace(pageLocation.UrlName))
             {
                 updater.ModelState.AddModelError(Prefix, nameof(pageLocation.UrlName), "A value is required for 'UrlName'");
+                urlNameIsValid = false;
+            }
+
+            if (urlNameIsValid && !Regex.IsMatch(pageLocation.UrlName!, UrlNamePattern))
```

Comment:
Could this be achieved by adding a `RegularExpression` data annotation attribute on the model?
(Genuinely don't know if it's supported!)

Reply:
Not sure, I'm just following the way they do validation within OC, so potentially not!